Siva - fix/create-and-improve-create-new-team

pavanputti opened the pull request:
# Description


<img width="664" height="462" alt="Screenshot 2026-02-14 at 9 18 40 PM" src="https://github.com/user-attachments/assets/d9ced14e-738d-454f-9fd4-078610ae3d4e" />


[video](https://www.loom.com/share/eeb23108293043cd9a39a898fc28c01c)

## Related PRS (if any):
This frontend PR is related to the backend PR [1340](https://github.com/OneCommunityGlobal/HGNRest/pull/1340)
 To test this frontend PR, you need to checkout the backend PR [1340.](https://github.com/OneCommunityGlobal/HGNRest/pull/1340)
…

## Main changes explained:
  Update Create New Team component:
    Fixed the bug preventing team creation by correcting the form submission logic and backend API integration.
    Implemented UX improvements to match the Profile > Teams > Assign Team > Create New Team workflow, including:
    Consistent popup form with team name and optional description fields.
    Real-time validation for empty or duplicate team names.
    Autocomplete feature showing matching existing teams and an option to create a new team.
    Non-existent team creation flow with "Create Team" and "Cancel team creation" options.
    Clear success or error messages upon form submission.
  …

## How to test:
1.Check out the current branch: git checkout fix/create-and-improve-create-new-team
2.Run npm install to install dependencies.
3.Start the app locally with npm start.
4.Clear site data/cache in the browser.
5.Log in as an Admin user or Owner (or a role with team creation permissions).
6.Navigate to Other Links > Team Management > Create New Team and test the following cases:
- Create: Enter a new name (e.g. Test Team), choose “Create new team: [name]” and confirm → team is created and appears in the list.
- Empty name: Submit with no name → “Please enter a team name” (or similar); no team created.
- Duplicate: Use an existing team name and try to create → message that team already exists.
- Autocomplete: Type (e.g. Te) → list of matches and “Create new team: [typed name]” option.
- Cancel: Open modal, type a name, then cancel/close → modal closes, no team created

## Screenshots or videos of changes:

Create New Team (Before Fix ):

https://github.com/user-attachments/assets/8f55075c-38e0-4e14-9d96-67639500d88c

Create New Team (After Fix ):

https://github.com/user-attachments/assets/e19f68f7-d76f-429b-99ea-7387bbe164f8

pavanputti committed:
Refactor Teams component to integrate AddTeamPopup and enhance team filtering logic. Removed CreateNewTeamPopup and updated state management for team data. Improved search functionality and UI elements in TeamTableSearchPanel and AddTeamsAutoComplete components.

pavanputti committed:
Refactor onSelectAssignTeam function in Teams component to remove unused parameter, streamlining the code for better readability and maintainability.

pavanputti committed:
Update TeamTableSearchPanel tests: remove debug statement and correct placeholder text for search input to enhance clarity and accuracy of test cases.

Saicharan1505 reviewed (dismissed):
Tested the updated “Create New Team” workflow under Other Links > Team Management. Verified bug fix and UX improvements including autocomplete, duplicate name validation, and non-existent team creation flow. All test cases (basic creation, empty name, duplicate name, autocomplete, cancel creation) worked as expected. The new team appears in the list and supports further actions like adding members or editing. No issues found.

![3658 #1](https://github.com/user-attachments/assets/df703376-d132-48b6-94de-ff7c8fda0573)
![3658 #2](https://github.com/user-attachments/assets/5a928b8c-3cfe-4525-971a-59546920f7a0)
![3658 #3](https://github.com/user-attachments/assets/01824973-38a3-4534-bddd-f9f2750e3b41)
![3658 #5](https://github.com/user-attachments/assets/b3c2cf26-eaeb-4f39-b237-471f20423028)
![3658 #4](https://github.com/user-attachments/assets/1201d2ec-45fc-4f7c-b952-2c41ea2ae000)

marcusyi1 reviewed (changes requested):
Everything else in this branch works as expected. However there is an edge case where if you put a space in the team name it still registers as a new team. Could be good idea to add something to prevent this.


https://github.com/user-attachments/assets/dec579ed-9b16-4e24-bfa1-09b2838125c0

uhakruthi1 reviewed (dismissed):
Everything is working well locally. Tested the frontend thoroughly with backend PR #1340. Team creation flow, validation, and autocomplete features are all functioning as expected.Verified team appears correctly in the list after creation and can be edited or managed as intended. UX behavior matches the described workflow smoothly.
<img width="1467" alt="Screenshot 2025-07-08 at 5 08 17 PM" src="https://github.com/user-attachments/assets/3afeebd9-648e-4cd7-ba91-c227c53e0c2c" />
<img width="1467" alt="Screenshot 2025-07-08 at 6 54 58 PM" src="https://github.com/user-attachments/assets/29085b26-27dc-48e2-b3fd-96eb6bafcd8b" />
<img width="1467" alt="Screenshot 2025-07-08 at 6 55 05 PM" src="https://github.com/user-attachments/assets/9d0dd1de-b5c0-43e6-8bf0-0ffd071ee15c" />
<img width="1467" alt="Screenshot 2025-07-08 at 6 56 56 PM" src="https://github.com/user-attachments/assets/c345afd3-b8d6-4250-bcbc-0608b6ff2f28" />
<img width="1467" alt="Screenshot 2025-07-08 at 7 07 10 PM" src="https://github.com/user-attachments/assets/b48adf04-15a8-4655-9611-28bb22b871d5" />
<img width="1467" alt="Screenshot 2025-07-08 at 7 07 20 PM" src="https://github.com/user-attachments/assets/ae79d188-c63a-4fec-bad6-b600175a51ae" />

Juhitha-Reddy commented:
Tested all scenarios and everything is working as expected.

<img width="1454" height="330" alt="image" src="https://github.com/user-attachments/assets/7736fd5d-938c-49c8-87a4-d8eb2530ff55" />
<img width="773" height="481" alt="image" src="https://github.com/user-attachments/assets/d7fe664c-d14a-41a4-ac97-bb736744e03b" />
<img width="1897" height="898" alt="image" src="https://github.com/user-attachments/assets/0893ed39-26b8-418d-b0f0-89816b634713" />
<img width="1901" height="719" alt="image" src="https://github.com/user-attachments/assets/6908abe2-a4a1-4302-b57e-ea896d2fb4ef" />

saikrishnaraoyadagiri reviewed (changes requested):
Hi, Reviewed this PR, all cases are working but when we get the error message showing team already exists it is staying in the popup even after clicking cancel team, create ream or confirm, until we refresh the browser. Attaching video for reference.

https://github.com/user-attachments/assets/81b436d6-262f-4d48-a99c-6ac739b55aac

harshavarma29 reviewed (changes requested):
I have tested the latest changes locally using the admin account with the following scenarios listed below:

1) Creating a new team.
2) Empty team name.
3) Autocomplete functionality.

They are working well. However, below two scenarios not working as expected.

1) Duplicate team name:
    When creating a duplicate team name, a pop-up with a message 'A team with this name already exists' is not appearing in the UI, even though it is handled in the code.

   ![existing-team-popup](https://github.com/user-attachments/assets/439672a5-d93c-401a-9e60-3a64ceb3e217)

     Please refer to the attached video for this scenario.
     
     https://github.com/user-attachments/assets/8ae0d7c4-0738-4a55-9bf0-d3bf34666bf0

2) Cancel team creation.
    When clicking on 'Cancel team creation' for a non-existing team, the pop-up does not getting disappear.

    https://github.com/user-attachments/assets/faa2b9c1-020c-451b-b347-976a581fae49

pavanputti committed:
Enhance AddTeamPopup and AddTeamsAutoComplete components: Implement duplicate team validation, improve search text handling, and update UI alerts for better user experience. Clear duplicate state on close and refine team creation logic.

harika-majji reviewed (changes requested):
Tested the functionality as descripted in description.
**Team List -  edit button is not working.** ( see last image for reference)


**1. Basic Team Creation:**

Team is created successfully, a success message appears, and the new team is listed under Team Management as an active team.

<img width="1424" height="781" alt="Screenshot 2025-07-21 at 11 09 50 AM" src="https://github.com/user-attachments/assets/a07520e8-89a3-4397-8834-b973a3afc1bc" />


<img width="1445" height="692" alt="Screenshot 2025-07-21 at 11 10 01 AM" src="https://github.com/user-attachments/assets/068e11e1-fb70-4097-9ec9-ebc94840adbe" />

**2. Empty Team Name:**
 Error message appears: "Please enter a team name."

<img width="1317" height="495" alt="Screenshot 2025-07-21 at 11 10 18 AM" src="https://github.com/user-attachments/assets/5109c083-41ae-48c2-b6d5-7e72de193762" />

**3. Duplicate Team Name:**
 Warning message appears about the team already existing.

<img width="1317" height="529" alt="Screenshot 2025-07-21 at 11 10 37 AM" src="https://github.com/user-attachments/assets/c179ebef-67f9-4176-b4a8-d042ae873997" />


**4. Autocomplete Functionality:** works as expected.

<img width="1147" height="477" alt="Screenshot 2025-07-21 at 11 10 53 AM" src="https://github.com/user-attachments/assets/10bea2ab-1e89-454b-a0bd-8a1df315a441" />

**5. Non-existent Team Creation Flow:**
Warning message when a team doesn't exists "Oops, this team does not exist!" with two buttons: "Create Team" and "Cancel team creation."

<img width="1034" height="447" alt="Screenshot 2025-07-21 at 11 11 41 AM" src="https://github.com/user-attachments/assets/94258ece-1857-455c-b235-a4ab92568462" />

**6. Cancel Team Creation:**
 Cancel button works as expected.


After creating a team:
The new team appears in the teams list.
- Delete Button is working as expected.
- Add/ Delete Members for a team works as expected.


**- The edit button is not working.**

<img width="1252" height="227" alt="Screenshot 2025-07-21 at 11 19 49 AM" src="https://github.com/user-attachments/assets/f45cd541-c520-4fb4-ad11-4d6c38d6cf77" />

sankargalla reviewed (changes requested):
I reviewed the PR, and all the following use cases work as expected: Basic Team Creation, Empty Team Name, Duplicate Team Name, Autocomplete Functionality, Non-existent Team Creation Flow, and Cancel Team Creation. After successfully creating a team, its details appear in the table with the Add Members, Edit, and Delete buttons. The Add Members and Delete actions function correctly.

**Observation:** Edit button is not working as expected.

<img width="1436" height="777" alt="PR 3658(new)" src="https://github.com/user-attachments/assets/b541c8de-d819-4b4d-bf6d-075e4c237141" />
<img width="500" height="335" alt="PR 3658(1)" src="https://github.com/user-attachments/assets/21478455-4617-4281-ae97-3181bab50b01" />
<img width="495" height="309" alt="PR 3658(2)" src="https://github.com/user-attachments/assets/375650ca-7e73-4d2e-8583-37435a17bb54" />
<img width="1434" height="776" alt="PR 3658(delete) " src="https://github.com/user-attachments/assets/4003fb4e-9d30-48ae-bc6d-1ef932b60481" />

pavanputti committed:
Implement team editing functionality in Teams component and enhance AddTeamPopup: Added state management for editing teams, including validation and UI updates for better user experience.

pavanputti committed:
Update AddTeamPopup component and tests: Handle null selectedTeam case in ID check, update test assertions for team name display, and improve loading state handling in button interactions.

pavanputti commented:
> Everything else in this branch works as expected. However there is an edge case where if you put a space in the team name it still registers as a new team. Could be good idea to add something to prevent this.

 @marcusyi1 Thanks for pointing that out! I've addressed the edge case where spaces in the team name were creating new teams. The logic now trims whitespace and checks for duplicates accordingly. I've committed the changes—please take a look and let me know if anything else needs adjustment.

pavanputti committed:
Enhance AddTeamPopup functionality: Implement team code generation logic, improve error handling during team updates, and refine loading state management. Update tests for button interactions to ensure proper loading state behavior.

BenithaSri reviewed (commented):
1. Basic Team Creation: 
Expected: Team is created successfully, a success message appears, and the new team is listed under Team Management as an active team: Successful!

However, it gets disappeared as soon as the toast alter is gone.

<img width="1512" height="982" alt="Screenshot 2025-07-30 at 11 43 23 AM" src="https://github.com/user-attachments/assets/61b55da8-5a7f-4012-b08c-dfed3c4d1fc2" />

<img width="1379" height="360" alt="Screenshot 2025-07-30 at 12 13 28 PM" src="https://github.com/user-attachments/assets/ae5635c9-8900-4f34-b615-8fab2db49c13" />

I am not able to find the team in search as well.

<img width="1373" height="479" alt="Screenshot 2025-07-30 at 12 13 58 PM" src="https://github.com/user-attachments/assets/00b3daf0-3c9c-42ce-a367-96446b39a861" />


2.Empty Team Name: 
Expected: Error message appears: "Please enter a team name." : Successful!

<img width="688" height="370" alt="Screenshot 2025-07-30 at 11 46 23 AM" src="https://github.com/user-attachments/assets/ede32558-56ac-4ad3-81e3-8764bbe1e906" />

3.Duplicate Team Name: 
Expected: Warning message appears about the team already existing. : Successful!
<img width="1251" height="405" alt="Screenshot 2025-07-30 at 11 47 49 AM" src="https://github.com/user-attachments/assets/05c590e7-8207-4b05-a843-952eee809a02" />

4. Autocomplete Functionality
Expected: Autocomplete shows a list of matching existing teams (if any) and an option to "Create new team: [typed name]." : Successful!
<img width="1356" height="408" alt="Screenshot 2025-07-30 at 11 49 42 AM" src="https://github.com/user-attachments/assets/d0f24dcd-ec60-45bf-b672-ceb0a1131b37" />

marcusyi1 reviewed (changes requested):
This branch is not supported by Node 20 anymore.

pavanputti committed:
Update Node version requirement to >=20 to match development branch

pavanputti committed:
Merge development branch: Resolve conflicts and update Node version to >=20

pavanputti committed:
Fix test and linting issues in AddTeamPopup test

pavanputti committed:
Fix linting errors in AddTeamPopup

pavanputti committed:
Implement useParams hook to extract activityId from URL

- Added useParams import from react-router-dom
- Extract activityid parameter from URL route
- Added validation and error handling for missing activityId
- Component now receives activityId from URL parameter
- Prepared component for dynamic data fetching in next commit

Anusha-Gali reviewed (changes requested):
Hi Pavan,

I tried reviewing your PR but most of the key successful messages have changed. Unable to see any duplicate message as well. There is no Cancel option as well.
<img width="1232" height="771" alt="Screenshot 2026-01-16 at 4 21 53 PM" src="https://github.com/user-attachments/assets/d8d0dbdf-ecb4-453d-ad1d-7a820fec4143" />
<img width="1232" height="771" alt="Screenshot 2026-01-16 at 4 23 54 PM" src="https://github.com/user-attachments/assets/ada7faaf-6822-44cf-8862-7d16acb6bb00" />
<img width="1232" height="771" alt="Screenshot 2026-01-16 at 4 24 06 PM" src="https://github.com/user-attachments/assets/66ed88ae-abd3-458e-80e2-a165a9ba1374" />
<img width="1232" height="771" alt="Screenshot 2026-01-16 at 4 24 31 PM" src="https://github.com/user-attachments/assets/1966f811-2782-4717-80ec-3929adc96ee7" />
<img width="1232" height="771" alt="Screenshot 2026-01-16 at 4 24 50 PM" src="https://github.com/user-attachments/assets/24e4cb52-b1e8-48dc-bf02-c838feb7a770" />

pavanputti committed:
Merge branch 'development' into fix/create-and-improve-create-new-team

pavanputti committed:
fix: resolve linting errors in Teams and AddTeamPopup components

Anusha-Gali reviewed (changes requested):
Hi Pavan,

I have re-reviewed your PR as there were few commits made. Thanks for fixing few issues however i was unable to verify the below:
- Autocomplete Functionality
- Non-existent Team Creation Flow
- Cancel Team Creation

**Verified**
<img width="1271" height="753" alt="Image" src="https://github.com/user-attachments/assets/12ba132a-5199-4e59-8f23-70b92902af54" />
<img width="1271" height="753" alt="Image" src="https://github.com/user-attachments/assets/127a4c44-bd8f-466a-925a-02daac68a36f" />
<img width="1271" height="753" alt="Image" src="https://github.com/user-attachments/assets/f6d9eac7-bcd0-4c07-886d-b5f161629d15" />
<img width="1271" height="753" alt="Image" src="https://github.com/user-attachments/assets/5fdaf4b5-10b0-4e7a-8622-5b9c3a3f8978" />
<img width="1271" height="753" alt="Image" src="https://github.com/user-attachments/assets/f5fbd8c8-2494-4512-83f3-cc68ece59438" />

pavanputti committed:
Handle whitespace-only team input.

Ensure empty or space-only input shows the pick-a-team message and avoid showing create-new options for blank text.

pavanputti committed:
Improve add-team validation states.

Keep duplicate and empty-name messaging in sync with input changes and close the modal on cancel.

pavanputti committed:
Update add-team empty-name test.

Expect the new empty-name validation message in the add-team popup.

pavanputti committed:
Fix add-team test: focus input and wait for dropdown option.

pavanputti committed:
Merge remote fix/create-and-improve-create-new-team; keep add-team validation and test fixes

pavanputti committed:
Fix AddTeamPopup: use DarkMode.module.css instead of missing DarkMode.css

pavanputti committed:
Fix SonarCloud issues: add PropTypes, optional chaining, reduce cognitive complexity

pavanputti committed:
Reduce AddTeamPopup complexity: extract generateValidTeamCode and handleEditInputChange

pavanputti commented:
@Anusha-Gali 
Hi Anusha,

I’ve made the necessary changes. Could you please review when you get a chance?

Thanks,
Pavan

Anusha-Gali reviewed (changes requested):
Hi Siva,

Thanks for fixing the issues, upon review all the functionalities work as per requirement in light mode but the UI in dark mode is a bit dull making it difficult to see the warning.

<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/19e5a014-fdf5-4486-8aed-f52735e87a98" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/e146cbba-5037-469f-8042-8a9192de6649" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/3469f620-1180-4c28-ba19-7bfb6300f07b" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/9c132279-67a8-4167-bc4e-b2946facc19e" />
**Requires UI fix**
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/2c247c62-ff7b-4a10-8743-5008b0f850f6" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/031fa93b-bfe6-49c0-805d-002d2c434faa" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/d9a00cbe-0f2f-478b-adb1-8a626377c2cf" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/a6b1fd30-7f28-429e-84dc-a54a671068d3" />
<img width="1279" height="759" alt="Image" src="https://github.com/user-attachments/assets/5d6ff1f0-d946-4441-8dc1-b4884f77981f" />

pavanputti committed:
Merge branch 'origin/development' into fix/create-and-improve-create-new-team and resolve conflicts

pavanputti committed:
Fix SonarCloud issues: remove unused searchText prop, remove getCachedTeamMembers import, reduce cognitive complexity and extract nested ternary in AddTeamPopup

pavanputti committed:
Fix failing test: update placeholder text from 'Search Teams' to 'Search Text' in TeamTableSearchPanel test

pavanputti committed:
Fix SonarQube issues: fix broken import path, remove dead code, remove console.error, remove unused state and props

pavanputti committed:
Resolve merge conflict in TeamTableSearchPanel.jsx: properly resolve raw markers, keeping PropTypes and alias improvements

pavanputti committed:
Fix AddTeamPopup UX for Create New Team vs Assign Team flows, and enhance Dark Mode Alert aesthetics

pavanputti committed:
Resolve SonarCloud issues: reduce cognitive complexity and handle exceptions in catch blocks

one-community commented:
Thank you all, merging!
